apps/desktop/src/lib/components/EditMode.svelte

```diff
@@ -195,6 +196,7 @@
 		modeServiceSaving = 'loading';
 
 		await modeService.saveEditAndReturnToWorkspace();
+		await modeService.awaitNotEditing();
```

Comment:
I find these method calls are quite treacherous as they as I have a feeling they can throw an exception. Then the button could be stuck in loading state.

If one of these fails the (backend) method itself is likely to leave the working tree/repo in an invalid state, so improving the frontend is certainly only a part of what it takes to be truly 'atomic' or deterministic.

Reply:
`saveEditAndReturnToWorkspace` could throw. `awaitNotEditing` is just creating a promise that resolves whenever the backend emits the fact that we're no longer in edit.

I'll add a little `try ... finally` that takes you out of the loading state if `xxxEditAndReturnToWorkspace` fails.